collection/collection.go

```diff
@@ -28,22 +28,208 @@ var (
+			})
+			if err != nil {
+				log.Debugw("updating dataset across all collections", "InitID", evt.InitID, "err", err)
+			}
+		}
+	case event.ETAutomationWorkflowStopped:
+		if evt, ok := e.Payload.(event.WorkflowStoppedEvent); ok {
+			err := sm.set.UpdateEverywhere(ctx, evt.InitID, func(vi *dsref.VersionInfo) {
+				vi.RunStatus = evt.Status
+			})
+			if err != nil {
+				log.Debugw("updating dataset across all collections", "InitID", evt.InitID, "err", err)
+			}
+		}
+	case event.ETAutomationWorkflowCreated:
+		if wf, ok := e.Payload.(workflow.Workflow); ok {
+			err := sm.set.UpdateEverywhere(ctx, wf.InitID, func(vi *dsref.VersionInfo) {
+				vi.WorkflowID = wf.WorkflowID()
+			})
+
+			if err != nil {
+				log.Debugw("updating dataset across all collections", "InitID", wf.InitID, "err", err)
+			}
+		}
+	case event.ETAutomationWorkflowRemoved:
+		if wf, ok := e.Payload.(workflow.Workflow); ok {
+			err := sm.set.UpdateEverywhere(ctx, wf.InitID, func(vi *dsref.VersionInfo) {
+				vi.WorkflowID = ""
+			})
+
+			if err != nil {
+				log.Debugw("updating dataset across all collections", "InitID", wf.InitID, "err", err)
+			}
+		}
+	case event.ETFSICreateLink:
+		if link, ok := e.Payload.(event.FSICreateLink); ok {
+			sm.set.UpdateEverywhere(ctx, link.InitID, func(vi *dsref.VersionInfo) {
+				vi.FSIPath = link.FSIPath
+			})
+		}
+	case event.ETFSIRemoveLink:
+		if change, ok := e.Payload.(event.FSIRemoveLink); ok {
+			sm.set.UpdateEverywhere(ctx, change.InitID, func(vi *dsref.VersionInfo) {
+				vi.FSIPath = ""
+			})
+		}
+	}
+	return nil
 }
 
-// WritableSet is an extension interface on Set that adds methods for
-// adding and removing items
-type WritableSet interface {
-	Set
-	Put(ctx context.Context, profileID profile.ID, items ...dsref.VersionInfo) error
-	Delete(ctx context.Context, profileID profile.ID, initIDs ...string) error
+// Set maintains lists of dataset information, called a collection, with each
+// list scoped to a user profile. A user's collection may consist of information
+// from datasets they have created and datasets added from other users.
+// Collections are the canonical source of truth for listing a user's datasets
+// in a qri instance. While a collection owns the list, the fields in a
+// collection item are cached values gathered from other subsystems, and must
+// be kept in sync as subsystems mutate their state
+type Set interface {
+	// List the collection of a single user
+	List(ctx context.Context, pid profile.ID, lp params.List) ([]dsref.VersionInfo, error)
+	// Get info about a single dataset in a single user's collection
+	Get(ctx context.Context, pid profile.ID, initID string) (dsref.VersionInfo, error)
+	// PutList adds a list of `dsref.VersionInfo`s to a user's collection
+	PutList(ctx context.Context, pid profile.ID, items []dsref.VersionInfo) error
+	// Add adds a single dataset to a user's collection
+	Add(ctx context.Context, pid profile.ID, add dsref.VersionInfo) error
```

Comment:
Not a fan of this change. Drop `PutList` and flip `Add` to accept a variadic arg:

```go
// Add adds one or more versionInfos to a user's collection
Add(ctx context.Context, pid profile.ID, add ...dsref.VersionInfo) error`
```

Add should batched into a single transaction under the hood, having more than one way to add a dataset means more than one place where bugs can arise, and adds one more requirement to implementers